### PR TITLE
Strongly type `context` field in OGM

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

Strongly type `context` field in OGM.

## Complexity

Complexity: Low

### CODE REVIEW

**Review by tbwiss — APPROVED** [review]

File: packages/graphql/tests/schema/issues/3439.test.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { printSchemaWithDirectives } from "@graphql-tools/utils";
 import { gql } from "graphql-tag";
 import { lexicographicSortSchema } from "graphql/utilities";
 import { Neo4jGraphQL } from "../../../src";
-import { TestSubscriptionsPlugin } from "../../utils/TestSubscriptionPlugin";
+import { TestSubscriptionsMechanism } from "../../utils/TestSubscriptionsMechanism";
```
Comment (tbwiss) [inline packages/graphql/tests/schema/issues/3439.test.ts:24]: @darrellwarde, isn't that suppose to be in another PR? This PR is about the OGM `context` type

Comment (darrellwarde) [reply packages/graphql/tests/schema/issues/3439.test.ts:24]: Yes, it's just a broken test from a merge from `dev`, just taking the opportunity to fix it. 🙃

Comment (tbwiss) [reply packages/graphql/tests/schema/issues/3439.test.ts:24]: fair enough 😄

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 34 | 43 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 70 | 90 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 43 | 62 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3157 | 3355 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2944 | 2930 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 50 | 72 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 55 | 57 | 2174 |
|  aggregations.DeeplyNestedCount | 13070331 | 13070329 | 4639 | 4591 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 148 | 150 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 118 | 130 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 178 | 205 | 3003 |
|  connections.Connection | 14082 | 14082 | 66 | 64 | 2174 |
|  connections.NestedConnection | 45472 | 45472 | 125 | 106 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 61 | 66 | 1040 |
|  connections.ConnectionWithSortAndCypher | 16325 | 16325 | 86 | 110 | 2174 |
|  create.SimpleMutation | 7 | 7 | 54 | 63 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 87 | 26 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 279 | 247 | 1040 |
|  delete.NestedDeleteInUpdate | 19198 | 19198 | 179 | 193 | 1183 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 76 | 58 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 92 | 131 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 45 | 45 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 54 | 62 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 23 | 30 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 45 | 46 | 2174 |
|  query.QueryWhere | 9697 | 9714 | 33 | 41 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9875 | 9892 | 48 | 64 | 2161 |
|  query.Nested | 10096041 | 10096041 | 10448 | 11813 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10808 | 11428 | 2004000 |
|  query.OrFilterOnRelationships | 42297 | 41980 | 204 | 229 | 1578 |
|  query.OrFilterOnRelationshipsAndNested | 35732 | 35267 | 207 | 246 | 1578 |
|  query.QueryWithNestedIn | 13336 | 13791 | 57 | 59 | 1195 |
|  query.NestedConnectionWhere | 9834 | 9834 | 57 | 73 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9946 | 9854 | 74 | 103 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19182 | 19044 | 124 | 195 | 1608 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 155 | 216 | 1134 |
|  query.Fulltext | 80 | 80 | 31 | 33 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 42 | 50 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 62 | 72 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14492 | 14546 | 86 | 96 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 49 | 63 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 95 | 106 | 4516 |
|  sorting.SortWithTopLevelCypher | 16162 | 16162 | 53 | 140 | 2174 |
|  unions.SimpleUnionQuery | 321 | 321 | 90 | 108 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 68 | 80 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 361 | 291 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 299 | 312 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 96 | 142 | 2002 |

</details>
        
Old Schema Generation: 38.526s
Schema Generation: 38.218s
Old Subgraph Schema Generation: 48.703s
Subgraph Schema Generation: 48.561s
